src/vm/prestub.cpp

```diff
@@ -1752,9 +1752,15 @@ PCODE MethodDesc::DoPrestub(MethodTable *pDispatchingMT)
     BOOL fWasPromotedToTier1 = FALSE;
     if (fEligibleForTieredCompilation)
     {
-        fEligibleForCallCounting = g_pConfig->TieredCompilation_CallCounting();
-        if (fEligibleForCallCounting)
+        if (RequestedAggressiveOptimization())
```

Comment:
How much slower can the call counting get with this call on the call counting path?

Note that `RequestedAggressiveOptimization` will access metadata that may include taking locks when the metadata are in R/W mode.

Reply:
I measured the call counting overhead in MusicStore, AllReady, and CscSource benchmarks, I don't see any noticeable difference with the changes. At the moment I think it's relatively easy to reduce the number of calls to `RequestedAggressiveOptimization()` per prestub invocation to a 2 - one where you highlighted above, and another in `TieredCompilationManager::GetJitFlags`. It would be possible to reduce that to 1 by plumbing the value through. Another simpler option may be to do what is done for `NoInlining`, which uses a bit in `MethodDescClassification` (`mdcNotInline`) to store that info in the MethodDesc once. It looks like some bits are available.

Reply:
> Another simpler option may be to do what is done for NoInlining, which uses a bit in MethodDescClassification (mdcNotInline)

I do not think it is worth burning a MethodDesc bit for this.